dts/bindings/dma/xilinx,eth-dma.yaml

```diff
@@ -0,0 +1,13 @@
+# Copyright 2024 CISPA Helmholtz Center for Information Security gGmbH
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Xilinx AXI DMA LogiCORE IP controller with compatibility string
```

Comment:
why can't the same string be used?

Reply:
Xilinx' Vitis tool can automatically generate device trees. This feature is often used as a starting point before further customization. The output generated by Vitis is context-dependent; for DMAs used together with the Ethernet subsystem, it generates an `eth-dma` compatible, for other uses it generates the other, more generic, compatible.
Linux actually binds different drivers for the DMA depending on the compatible: the entirety of the Ethernet subsystem including DMA has one driver, and there is an independent driver for the stand-alone DMA (e.g., useful for custom IPs connected to the DMA).

Reply:
Small correction: Upstream Linux actually does not actively *use* the `eth-dma` compatible. Instead, the Ethernet driver resolves the dma using the *axistream-connected* handle. Still, using the `eth-dma` handle prevents the generic DMA driver from claiming the device. 
References:
- Ethernet driver: https://github.com/torvalds/linux/blob/master/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
- DMA driver: https://github.com/torvalds/linux/blob/master/drivers/dma/xilinx/xilinx_dma.c
- My other PR https://github.com/zephyrproject-rtos/zephyr/pull/77732 has an exemplary device tree based on what Vitis generated automatically (notice the `eth-dma` compatible): https://github.com/zephyrproject-rtos/zephyr/blob/eda4bedfb96b2f07dd65142dc335d247e680b971/dts/riscv/openhwgroup/cv64a6.dtsi